### ISSUE TITLE
Porting the repository to a single unified CMake build structure

### ISSUE THREAD

**Issue by thatrobotguy** [PacktPublishing/Learn-CUDA-Programming#4]
# TL;DR
Please port the code to use `CMake` exclusively: code compilation is right now a bear to deal with.

# Long story:

I have been poking around this code repository trying out samples and I have run into multiple issues getting examples to compile. This repository would be a million times cleaner if all of the code was compiled by a single `cmake` command in the root directory. I should also mention that mixing of cuda and non-cuda projects often occurs through the use of the CMake build tools (ROS is a big one) and this code would be more usable and portable (say, to Windows) if CMake handled all of the code compilation. I was going to convert it myself to CMake (which looks to be a monumental task looking at all of the files) but there is so much broken and/or not-compiling code I would not know where to start.

I should say that I bought the CMake cookbook from Packt, and I think that this book needs to use the principles mentioned in that book to make this repository usable. This would also allow for a single folder to have all of the compiled files so that the `.gitinore` file can ignore them and therefore allow to more easily create pull requests with this repository should changes need to be made (as GCC and Ubuntu get updated, etc.). I should mention that I have made pull requests with other authors' repositories simply because the repository is so well organized it took me 10 minutes to do it, so I would recommend that this repo receive the same TLC.

I would normally call this a feature request, but the code is so broken it would solve a lot of  problems if this repo used CMake and Make together rather than just Make by itself.

**Comment by packt-pradeeps** [comment]
The authors have mentioned that a single CMake file is not possible because there are nearly 90 different applications. The CMake files have been updated as per the latest update of CUDA.
